app/model/SubscriptionAcquisitionComponents.scala

```diff
@@ -0,0 +1,181 @@
+package model
+
+import com.gu.acquisition.model.{OphanIds, ReferrerAcquisitionData}
+import com.gu.acquisition.typeclasses.AcquisitionSubmissionBuilder
+import com.gu.memsub.Benefit.{PaperDay, Weekly}
+import com.gu.memsub.BillingPeriod
+import com.gu.memsub.BillingPeriod.{Month, Quarter, SixMonths, Year}
+import com.gu.memsub.promo.{PercentDiscount, Promotion}
+import com.gu.stripe.Stripe.Charge
+import com.typesafe.scalalogging.LazyLogging
+import forms.SubscriptionsForm
+import ophan.thrift.event.PaymentProvider.{Gocardless, Stripe}
+import ophan.thrift.event._
+import play.api.libs.json._
+
+import scala.collection.Set
+
+//import com.gu.acquisition.model.ReferrerAcquisitionData.referrerAcquisitionDataReads
+
+case class SubscriptionAcquisitionComponents(
+  subscribeRequest: SubscribeRequest,
+  promotion: Option[Promotion.AnyPromotion],
+  acquisitionDataJSON: Option[String]
+)
+
+
+object SubscriptionAcquisitionComponents {
+
+  implicit object subscriptionAcquisitionSubmissionBuilder
+    extends AcquisitionSubmissionBuilder[SubscriptionAcquisitionComponents] with LazyLogging {
+
+    def buildOphanIds(components: SubscriptionAcquisitionComponents): Either[String, OphanIds] = {
+      import components._
+
+      Right(
+        OphanIds(
+          pageviewId = subscribeRequest.genericData.ophanPageViewId,
+          visitId = subscribeRequest.genericData.ophanVisitId,
+          browserId = subscribeRequest.genericData.ophanBrowserId
+        )
+      )
+    }
+
+    private def parseJsonSafely(json: String): Option[JsValue] = {
+      try {
+        Some(Json.parse(json))
+      } catch {
+        case scala.util.control.NonFatal(t) =>
```

Comment:
why not just wrap this in the Try {...} as it catches the same thing.  you can do an onFailure to log and then a toOption